memory/src/main/java/org/opensearch/ml/memory/action/conversation/CreateInteractionTransportAction.java

```diff
@@ -93,10 +96,9 @@ protected void doExecute(Task task, CreateInteractionRequest request, ActionList
         Integer traceNumber = request.getTraceNumber();
         try (ThreadContext.StoredContext context = client.threadPool().getThreadContext().newStoredContext(true)) {
             ActionListener<CreateInteractionResponse> internalListener = ActionListener.runBefore(actionListener, () -> context.restore());
-            ActionListener<String> al = ActionListener
-                .wrap(iid -> { internalListener.onResponse(new CreateInteractionResponse(iid)); }, e -> {
-                    internalListener.onFailure(e);
-                });
+            ActionListener<String> al = ActionListener.wrap(iid -> {
+                cmHandler.updateConversation(cid, new HashMap<>(), getUpdateResponseListener(cid, iid, internalListener));
+            }, e -> { internalListener.onFailure(e); });
             if (parintIid == null || traceNumber == null) {
```

Comment:
[nit] what is `parintIid`? I guess this is parent id of something. Parent id of what?

Reply:
parent Id is for traces. parent id is null for final interactions. traces use this parent id to find the final interaction.